spec/src/indexSpec.js

```diff
@@ -136,6 +136,42 @@ describe('cloudwatch-metrics', function() {
       metric.put(1, 'metricName', [{Name:'ExtraDimension',Value: 'Value'}]);
       metric.put(2, 'metricName', [{Name:'ExtraDimension',Value: 'Value'}]);
     });
+
+    it('should set a Timestamp if specified in the options', function(done) {
+      attachHook(function(data, cb) {
+        expect(data).toEqual({
+          MetricData: [{
+            Dimensions: [{
+              Name: 'environment',
+              Value: 'PROD'
+            }, {
+              Name: 'ExtraDimension',
+              Value: 'Value'
+            }],
+            MetricName: 'metricName',
+            Unit: 'Count',
+            Timestamp: jasmine.any(String),
+            Value: 1
+          }],
+          Namespace: 'namespace'
+        });
+        expect(Date.parse(data.MetricData[0].Timestamp)).toBeLessThan(Date.now());
```

Comment:
millisecond precision affords for this to be the same as the timestamp
```suggestion
        expect(Date.parse(data.MetricData[0].Timestamp)).toBeLessThanOrEqual(Date.now());
```